traitsui/testing/tester/query.py

```diff
@@ -16,6 +16,30 @@
 """
 
 
+class Selected:
+    """ An object representing an interaction to obtain the object which is
+    currently selected.
+
+    Implementations should return the selected object, or None if nothing is
+    selected.
+    """
+    pass
+
+
+class SelectedIndices:
+    """ An object representing an interaction to obtain the indices of objects
+    which are currently selected.
+
+    Implementations should return either a list of indicies of the selected
+    objects, a single index of a lone selected object, or -1 if nothing is
+    selected.
+
+    Note that an index could be an integer, or it could be a tuple e.g.
+    corresponding to a specific cell at some (row, column) in a table.
```

Comment:
under what circumstances do we return an integer vs a tuple.

Reply:
Currently, it would be an int if selecting entire rows or columns, whereas it would be a tuple if selecting specific cells.  However, if used with say a ListEditor, InstanceEditor, or EnumEditor, a selected index would likely be an integer.  Currently those implementations do not exist (in current tests querying what is selected we just use `SelectedText` rather than inspecting the object itself that is selected).  These implementations can be added in the future though.
I've updated the doc string to try to make this more clear.